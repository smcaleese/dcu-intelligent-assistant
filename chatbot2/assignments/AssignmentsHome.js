import React, { useEffect, useState } from 'react';
import { View, Text, TextInput, TouchableOpacity, Image, ScrollView } from 'react-native';
import arrowImage from '../assets/arrow.png';
import { Delete } from 'react-native-feather';
import styles from './AssignmentsHomeStyle';

function CustomDropdown(props) {
    return (
        <TouchableOpacity style={[styles.dropdown, props.style]} activeOpacity={0.95} onPress={() => props.onPress()}>
            <Text style={styles.dropdownText}>{props.text}</Text>
            <Image source={arrowImage} style={styles.arrowImage} />
        </TouchableOpacity>
    )
}

export default function AssignmentsHome(props) {

    const [inputValue, setInputValue] = useState('')
    const [dateValue, setDateValue] = useState('')
    const [showAssignmentDropDown, setShowAssignmentDropDown] = useState(true)
    const [showUserDropDown, setShowUserDropDown] = useState(false)

    let assignments = []
    // assignments from assignments page:
    if(typeof props.assignments !== "undefined") {
        assignments = props.assignments
    }
    // assignments from chatbot:
    else if(typeof props.route.params !== "undefined") {
        assignments = props.route.params.assignments
    }

    const onSubmitAssignment = async () => {
        if (inputValue.length > 0 && dateValue.length > 0) {
            await props.handleAddAssignment(props.email, inputValue, dateValue)
            await props.handleGetAssignments(props.email)
        }
    }

    const onDeleteAssignment = async (assignment) => {
        await props.handleDeleteAssignment(assignment)
        await props.handleGetAssignments(props.email)
    }

    const assignmentBoxes = assignments.map((assignment, index) => {

        let background
        if (index % 2 === 0) {
            background = "#EFEFEF"
        } else {
            background = "white"
        }

        return (
            <View style={[styles.assignmentBox, { backgroundColor: background }]} key={index}>
                <Text style={styles.assignmentText}>{assignment["date"]}, {assignment["name"]}</Text>
                <Delete height={30} width={30} stroke='#f00'
                    style={{margin: 10}} onPress={() => onDeleteAssignment(assignment)} />
            </View>
        )
    })

    const handleLogout = () => {
        props.setLoggedIn(false)
        props.setAutoLogin(false)
    }

    const onDeleteAccount = () => {
        props.onDeleteAccount()
    }

    return (
        // create custom dropdown component

        <ScrollView style={styles.container} contentContainerStyle={{ alignItems: 'center' }}>
            <View style={styles.titleBox}>
                <Text style={styles.titleText}>Assignments</Text>
            </View>
            <View style={styles.assignmentsContainer}>
                {assignmentBoxes}
            </View>
            <CustomDropdown text="Add Assignment" onPress={() => setShowAssignmentDropDown(state => !state)} />
            { showAssignmentDropDown &&
                <View style={styles.dropdownBox}>
                    <TextInput
                        style={[styles.textInput, {marginTop: 30}]}
                        onChangeText={text => setInputValue(text)}
                        value={inputValue}
                        placeholder="Assignment Name"
                    />
                    <TextInput
                        style={styles.textInput}
                        onChangeText={text => setDateValue(text)}
                        value={dateValue}
                        placeholder="Date"
                    />
                    <TouchableOpacity 
                        style={styles.customButton}
                        activeOpacity={0.9} onPress={() => onSubmitAssignment()}>
                        <Text style={styles.customButtonText}>Submit</Text>
                    </TouchableOpacity>
                </View>
            }
            <CustomDropdown text="User Settings" onPress={() => setShowUserDropDown(state => !state)} />
            { showUserDropDown &&
                <View style={[styles.dropdownBox, {marginBottom: 24}]}>
                    <TouchableOpacity style={[styles.customButton, styles.userDropDownButton, {marginTop: 24}]}
                        activeOpacity={0.9} onPress={() => handleLogout()}>
                        <Text style={styles.customButtonText}>Log Out</Text>
                    </TouchableOpacity>
                    <TouchableOpacity style={[styles.customButton, styles.userDropDownButton, {backgroundColor: '#FF5000'}]}
                        activeOpacity={0.9} onPress={() => onDeleteAccount()}>
                        <Text style={styles.customButtonText}>Delete Account</Text>
                    </TouchableOpacity>
                </View>
            }
        </ScrollView>
    )
}
