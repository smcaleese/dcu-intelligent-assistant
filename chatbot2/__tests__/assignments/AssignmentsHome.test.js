import React from 'react';
import {create} from 'react-test-renderer';
import AssignmentsHome from '../../assignments/AssignmentsHome';

const screen = create(<AssignmentsHome assignments={[]} />)

test('AssignmentsHome Snapshot Test', () => {
    expect(screen).toMatchSnapshot()
})