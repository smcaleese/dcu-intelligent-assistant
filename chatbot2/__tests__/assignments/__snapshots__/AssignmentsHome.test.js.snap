// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssignmentsHome Snapshot Test 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
    }
  }
  style={
    Object {
      "backgroundColor": "#DEDEDE",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "white",
          "marginTop": 24,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 20,
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        Assignments
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "backgroundColor": "#4169e1",
          "width": "80%",
        }
      }
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4269E1",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 32,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 16,
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        Add Assignment
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/arrow.png",
          }
        }
        style={
          Object {
            "height": 30,
            "margin": 16,
            "width": 30,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "width": "80%",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Assignment Name"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderColor": "gray",
              "borderRadius": 5,
              "borderWidth": 1,
              "margin": 16,
              "paddingLeft": 15,
              "width": "80%",
            },
            Object {
              "marginTop": 30,
            },
          ]
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Date"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "white",
            "borderColor": "gray",
            "borderRadius": 5,
            "borderWidth": 1,
            "margin": 16,
            "paddingLeft": 15,
            "width": "80%",
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#4269E1",
            "borderRadius": 5,
            "margin": 10,
            "marginBottom": 20,
            "opacity": 1,
            "paddingLeft": 10,
            "paddingRight": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "padding": 16,
              "textAlign": "center",
            }
          }
        >
          Submit
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4269E1",
          "borderRadius": 5,
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 32,
          "opacity": 1,
          "width": "80%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "flex": 1,
            "fontSize": 16,
            "padding": 16,
            "textAlign": "center",
          }
        }
      >
        User Settings
      </Text>
      <Image
        source={
          Object {
            "testUri": "../../../assets/arrow.png",
          }
        }
        style={
          Object {
            "height": 30,
            "margin": 16,
            "width": 30,
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
